src/components/tabs/ScatterPlotTab.tsx

```diff
@@ -0,0 +1,705 @@
+  const selectedTrack = useTransitionedDebounce(props.selectedTrack, startTransition, () => setIsRendering(true), 0);
+  const currentFrame = useTransitionedDebounce(props.currentFrame, startTransition, () => setIsRendering(true), 0);
+
+  // TODO: Implement color ramps in a worker thread. This was originally removed for performance issues.
+  // Plotly's performance can be improved by generating traces for colors rather than feeding it the raw
+  // data and asking it to colorize it.
+  // https://www.somesolvedproblems.com/2020/03/improving-plotly-performance-coloring.html
+
+  // Wrap changes to axis and range in a transition to prevent blocking the UI thread.
+  const [rangeType, _setRangeType] = useState<RangeType>(DEFAULT_RANGE_TYPE);
+  const setRangeType = (newRangeType: RangeType): void => {
+    if (newRangeType === rangeType) {
+      return;
+    }
+    setIsRendering(true);
+    startTransition(() => {
+      _setRangeType(newRangeType);
+    });
+  };
+
+  const [xAxisFeatureName, _setXAxisFeatureName] = useState<string | null>(null);
+  const setXAxisFeatureName = (newFeatureName: string | null): void => {
+    if (newFeatureName === xAxisFeatureName) {
+      return;
+    }
+    setIsRendering(true);
+    startTransition(() => {
+      _setXAxisFeatureName(newFeatureName);
+    });
+  };
+  const [yAxisFeatureName, _setYAxisFeatureName] = useState<string | null>(null);
+  const setYAxisFeatureName = (newFeatureName: string | null): void => {
+    if (newFeatureName === yAxisFeatureName) {
+      return;
+    }
+    setIsRendering(true);
+    startTransition(() => {
+      _setYAxisFeatureName(newFeatureName);
+    });
+  };
+
+  //////////////////////////////////
+  // Click Handlers
+  //////////////////////////////////
+
+  /**
+   * Returns true if a Plotly mouse event took place over a histogram subplot.
+   */
+  function isHistogramEvent(eventData: Plotly.PlotMouseEvent): boolean {
+    return eventData.points.length > 0 && eventData.points[0].data.type === "histogram";
+  }
+
+  // Add click event listeners to the plot. When clicking a point, find the track and jump to
+  // that frame.
+  const onClickPlot = useCallback(
+    (eventData: Plotly.PlotMouseEvent): void => {
+      if (!dataset) {
+        return;
+      }
+      if (eventData.points.length === 0 || isHistogramEvent(eventData)) {
+        // User clicked on nothing or on a histogram
+        props.findTrack(null, false); // clears tracks. TODO: Is this expected behavior?
+        return;
+      }
+
+      const point = eventData.points[0];
+      const objectId = pointNumToObjectId.current[point.pointNumber];
+      console.log("Point number: " + point.pointNumber + " Object ID: " + objectId);
+      const trackId = dataset.getTrackId(objectId);
+      const frame = dataset.times ? dataset.times[objectId] : undefined;
+      if (frame !== undefined) {
+        props.setFrame(frame).then(() => {
+          props.findTrack(trackId, false);
+        });
```

Comment:
Try using other properties of Plotly Datum to include additional information (try using i, j, or k fields, or customdata?)